toolkit/tools/internal/pkgjson/pkgjson.go

```diff
@@ -339,78 +338,56 @@ func (interval *PackageVerInterval) Compare(other *PackageVerInterval) (result i
 	return equalTo
 }
 
-// versionInInterval returns true if ver is a version in the set of versions represented in interval
-func (interval *PackageVerInterval) versionInInterval(ver *versioncompare.TolerantVersion) (valid bool) {
-	if ver.Compare(interval.LowerBound) == 0 && interval.LowerInclusive {
-		return true
-	}
-	if ver.Compare(interval.UpperBound) == 0 && interval.UpperInclusive {
-		return true
-	}
-	if ver.Compare(interval.UpperBound) < 0 && ver.Compare(interval.LowerBound) > 0 {
-		return true
+// Contains checks if the interval fully contains the query range
+func (interval *PackageVerInterval) contains(queryInterval *PackageVerInterval) (contains bool) {
+	lowerBoundValid := false
+	upperBoundValid := false
+
+	// Check if the each bound of the query is within the interval, or if it is the same as the bound of the interval
+	// and the bound is inclusive.
+	if interval.LowerBound.LT(queryInterval.LowerBound) ||
+		(interval.LowerBound.EQ(queryInterval.LowerBound) && interval.LowerInclusive) ||
+		(interval.LowerBound.EQ(queryInterval.LowerBound) && (interval.LowerInclusive == queryInterval.LowerInclusive)) {
```

Comment:
Potential simplification (assuming I didn't mess something up):
```suggestion
		(interval.LowerBound.EQ(queryInterval.LowerBound) && (interval.LowerInclusive || !queryInterval.LowerInclusive)) {
```